packages/astro/src/astro-components/control/ProtectCSR.astro

```diff
@@ -0,0 +1,20 @@
+---
+import type { ProtectProps } from '../../types';
+
+type Props = Omit<ProtectProps, 'condition'>;
+
+const { role, permission } = Astro.props;
+---
+
+<clerk-protect
+  data-role={role}
+  data-permission={permission}
+>
+  <!-- TODO: Can't have a data attribute in a Fragment. Remove div wrapper later. -->
+  <div data-default-slot>
```

Comment:
Do we need some type of clerk specific prefix here ?
```suggestion
  <div data-clerk-control-slot-default>
```

Reply:
Addressed this, thanks!